src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import React, { useCallback, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todos,
+  setTodos,
+}) => {
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+  const [editTitle, setEditTitle] = useState(false);
+
+  const onCompleted = useCallback(() => {
+    const newTodos = [...todos];
+    const changeTodo = newTodos.find(newTodo => newTodo.id === todo.id);
+
+    if (changeTodo !== undefined) {
+      changeTodo.completed = !changeTodo.completed;
+    }
+
+    setTodos(newTodos);
+  }, [todos]);
```

Comment:
Avoid passing the props like todos and setTodos into single todo.
You should keep single responsibility principe, so in this case `TodoItem` should only be aware of certain Todo and work only with it.
Solution: 
- move `onCompleted(id: number)` out of TodoItem and pass it as props callback
- same for` onDeleteTodo `
- same for onEditTitleOnBlur (this function is accepting prop like todoTitle)
- and so on